src/KOKKOS/pair_snap_kokkos.h

```diff
@@ -13,9 +13,15 @@
 
 #ifdef PAIR_CLASS
 
-PairStyle(snap/kk,PairSNAPKokkos<LMPDeviceType>)
-PairStyle(snap/kk/device,PairSNAPKokkos<LMPDeviceType>)
-PairStyle(snap/kk/host,PairSNAPKokkos<LMPHostType>)
+#include "kokkos_type.h"
```

Comment:
This include statement breaks compilation of LAMMPS when KOKKOS is **not** enabled.
All `FeatureStyle(style/name,ClassName)` macros must be compatible to compile without any packages installed and without any references to package specific code, since there is a list of them compiled into the LAMMPS class in order to provide error messages for which package a style is provided by in case it is used but the package is not included into the LAMMPS executable in use.

Reply:
Thank you for explaining this @akohlmey , I was a bit confused. I'll figure out a workaround momentarily.

Reply:
The code failing is the `LAMMPS::init_pkg_lists()` function. As you may be able to see from the code, the ClassName macros argument is ignored in this function, so a simple `#ifdef LMP_KOKKOS` wrapper around the `#include` may be a sufficient solution.

Reply:
Ah, thank you for a solution suggestion---it's something I'll keep in my "back pocket".

In this case here I can remove the `#include kokkos_type.h` outright---it's a sloppy artifact of some intermediate attempts to use (for example) `PairSNAPKokkos<LMPDeviceType, SNAP_KOKKOS_REAL, SNAP_KOKKOS_DEVICE_VECLEN>`, which depended on macros in `kokkos_type.h`, before I switched to using a thin wrapper instead: https://github.com/weinbe2/lammps/blob/fea-snap-ui-aosoa/src/KOKKOS/pair_snap_kokkos.h#L261. `SNAP_KOKKOS_REAL` and `SNAP_KOKKOS_DEVICE_VECLEN`, and the associated template parameters to `PairSNAPKokkos`, are new with this PR, which is why these types of issues hadn't come up in the past.

I'll push the fix in a bit, just doing some due-diligence tests on my end first. This is an example of what happens when I automate my own testing a little too much!

Reply:
No worries, that is why we try to have a wide matrix of compilation tests for different platforms and different compilation setting on all pull requests. Too many compilation issues went past us in the past and even though it would not be feasible to go over all possible permutations of compilation settings with both supported build systems, we cover quite a lot of ground with the existing choices and those have been extremely helpful to identify potential portability issues and compilation failures.
https://github.com/lammps/lammps-testing/blob/master/scripts/builds/test_coverage.rst

Reply:
Another option (for the future) is to use the `accelerator_kokkos.h` wrapper, which has dummy implementations of Kokkos for when Kokkos isn't enabled.